airbyte-webapp/src/pages/source/SourceOverviewPage/sourceOverviewContext.ts

```diff
@@ -0,0 +1,25 @@
+import { useOutletContext } from "react-router-dom";
+
+import { SourceDefinitionRead, SourceRead, WebBackendConnectionListItem } from "core/request/AirbyteClient";
+import { useConnectionList } from "hooks/services/useConnectionHook";
+import { useGetSource } from "hooks/services/useSourceHook";
+import { useSourceDefinition } from "services/connector/SourceDefinitionService";
+
+interface SourceOverviewContext {
+  source: SourceRead;
+  sourceDefinition: SourceDefinitionRead;
+  connections: WebBackendConnectionListItem[];
+}
+
+export const useSourceOverviewContext = () => {
+  return useOutletContext<SourceOverviewContext>();
+};
+
+export const useSetupSourceOverviewContext: (id: string) => SourceOverviewContext = (id: string) => {
```

Comment:
:art: Code style nitpick: We most often specify the return type on the arrow function, so it reads a bit shorter.

```suggestion
export const useSetupSourceOverviewContext = (id: string): SourceOverviewContext => {
```

Reply:
Did this locally incase of lint issue.